provider/execution/execution.go

```diff
@@ -286,9 +269,13 @@ func (e TableExecutor) callTableResolve(ctx context.Context, client schema.Clien
 	if parent == nil {
 		e.Logger.Info("fetched successfully", "count", nc)
 	}
-	if err := e.cleanupStaleData(ctx, client, parent); err != nil {
-		return nc, diags.Add(fromError(err, diag.WithType(diag.DATABASE), diag.WithSummary("failed to cleanup stale data on table %q", e.Table.Name)))
+
+	if !diags.HasErrors() {
```

Comment:
I have a strong gut feeling we should avoid this change for now:

- Even a single column (e.g. in a child relation) that failed to be resolved can cause "error" diags, right? So a single column with a null value in any child relation will cause failure of cleanup .

Seems dangerous in any case. I'd really avoid doing this unless we have a specific reason to do it (e.g. customers complining, sentry errors, ....).

Reply:
if the column resolver returns error diags, it doesn't continue to complete all other tasks so (if there are any) the new rows are incomplete anyway. no postresourceresolver called, no internal resolvers (cq_id!) called. https://github.com/cloudquery/cq-provider-sdk/blob/main/provider/execution/execution.go#L367-L370

not sure if it will lead to more rows or not on edge cases.